src/GovernorCountingFractional.sol

```diff
@@ -134,23 +146,28 @@ abstract contract GovernorCountingFractional is Governor {
      */
     function _countVoteFractional(
         uint256 proposalId,
-        uint256 weight,
+        address account,
+        uint128 weight,
         bytes memory voteData
     ) internal {
         require(voteData.length == 48, "GovernorCountingFractional: invalid voteData");
 
-        (uint128 forVotes, uint128 againstVotes, uint128 abstainVotes) = _decodePackedVotes(voteData);
+        (uint128 _forVotes, uint128 _againstVotes, uint128 _abstainVotes) = _decodePackedVotes(voteData);
+
+        uint128 _existingWeight = _proposalVotersWeightCast[proposalId][account];
 
         require(
-            uint256(forVotes) + againstVotes + abstainVotes <= uint128(weight),
-            "GovernorCountingFractional: votes exceed weight"
+            uint256(_forVotes) + _againstVotes + _abstainVotes + _existingWeight <= weight,
+            "GovernorCountingFractional: vote would exceed weight"
         );
 
+        _proposalVotersWeightCast[proposalId][account] = _existingWeight + _forVotes + _againstVotes + _abstainVotes;
```

Comment:
Could save that sum to a var instead of computing it twice

Reply:
Thanks, nice catch!

Reply:
Done in 575f596